dialogue-blocking-channels/src/main/java/com/palantir/dialogue/blocking/DefaultCallingThreadExecutor.java

```diff
@@ -54,49 +52,44 @@
     }
 
     @Override
-    public ListenableFuture<?> submit(Runnable task) {
-        return queue.submit(task);
+    public void submit(Runnable task) {
+        queue.submit(task);
     }
 
     @Override
     public void executeQueue(ListenableFuture<?> await) {
         Preconditions.checkState(Thread.currentThread().getId() == threadId, "Executing queue on different thread");
-        await.addListener(() -> queue.submitNotifier(queue::poison), DialogueFutures.safeDirectExecutor());
+        DialogueFutures.addDirectListener(await, () -> queue.submitNotifier(queue::poison));
         try {
-            RunnableFuture<?> toRun;
+            Runnable toRun;
             while ((toRun = queue.getWork()) != null) {
-                toRun.run();
+                try {
+                    toRun.run();
+                } catch (Throwable t) {
+                    log.error("Failed to execute runnable {}", toRun, t);
+                }
             }
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
 
-            abortQueue();
+            queue.poison();
```

Comment:
Why do you think queue aborts are not necessary?

Reply:
I think that's right, I haven't thought about that.

Reply:
Calling cancel on the FutureTask instances is equivalent to ignoring the queue. We can get away with it here because we're on the only thread that can drain the queue. I'm planning to add comments here describing the constraints.